core/lib/merkle_tree/src/domain.rs

```diff
@@ -180,15 +189,20 @@ impl ZkSyncTree {
         let starting_leaf_count = self.tree.latest_root().leaf_count();
         let starting_root_hash = self.tree.latest_root_hash();
 
+        let instructions_with_hashed_keys = instructions
```

Comment:
Bikeshedding: It would be moderately easy to have `Self::transform_logs()` return `(Key, _)` instead of `(StorageKey, _)` (the original `StorageKey` can be easily associated with entries given that `transform_logs` doesn't skip or reorder input logs. OTOH, this would make it inconsistent with `Self::filter_write_logs()`, so I'm not insisting.

Reply:
the typing was changed to leverage the the additional address field, in addition to key, for constructing the `StateDiffRecord`